integration_test/common/common_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration_test
+
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/go-playground/validator/v10"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	//relative path to testdata folder
+	testdataDir   = "testdata"
+	inputYamlName = "input.yaml"
+)
+
+type testCase struct {
+	dirName        string
+	expectedErrors map[fieldError]struct{}
+}
+
+type fieldError struct {
+	field string
+	tag   string
+}
+
+func getYaml(t *testing.T, dirName string) []byte {
+	dir := path.Join(testdataDir, dirName, inputYamlName)
+	contents, err := ioutil.ReadFile(dir)
+	if err != nil {
+		t.Fatal("could not read dirName: " + dir)
+	}
+	str := strings.ReplaceAll(string(contents), "\r\n", "\n")
+
+	return []byte(str)
+}
+
+func UnmarshallAndValidate(t *testing.T, bytes []byte, i interface{}) error {
+	v := validator.New()
+	err := yaml.UnmarshalStrict(bytes, i)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return v.Struct(i)
+}
+
+func TestAll(t *testing.T) {
+	table := []testCase{
+		{
+			"pass",
```

Comment:
I think we should put the struct field names here i.e. `dirName: "pass"` but for every field in this slice